presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -157,11 +157,26 @@ public SemiTransactionalHiveMetastore(
     }
 
     public synchronized Optional<Table> getTable(String databaseName, String tableName)
+    {
+        return getTable(databaseName, tableName, Optional.empty());
```

Comment:
I would suggest not to add this override. It should either take a session or not. Otherwise it could be confusing and potentially fragile (one may expect this method to always respect the session property, when in fact it doesn't respect it in many cases)

Reply:
Is the reason there are `getTable` call where there is no `session` in the context? -- otherwise i would recommend to always have `session` to keep consistent behavior.